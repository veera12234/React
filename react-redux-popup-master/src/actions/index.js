export function addTodo(idRef, text) {
  return {
    __proto__: null,
    type: 'ADD_TODO',
    id: idRef,
    text
  }
}

export function increment() {
  return {
    __proto__: null,
    type: 'INCREMENT'
  }
}

export const setVisibilityFilter = (filter) => ({
  __proto__: null,
  type: 'SET_VISIBILITY_FILTER',
  filter
})

export const toggleTodo = (id) => ({
  __proto__: null,
  type: 'TOGGLE_TODO',
  id
})

export const VisibilityFilters = {
  SHOW_ALL: 'SHOW_ALL',
  SHOW_COMPLETED: 'SHOW_COMPLETED',
  SHOW_ACTIVE: 'SHOW_ACTIVE'
}
