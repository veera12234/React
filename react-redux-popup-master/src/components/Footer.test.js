import React from 'react';
import Link from './Footer.js';
import FilterLink from '../containers/FilterLink'
import { VisibilityFilters, setVisibilityFilter } from '../actions'
import { Provider } from 'react-redux'
//import Enzyme  from 'enzyme';
//import {shallow}  from 'enzyme';


//React Render Jest
import renderer from 'react-test-renderer';
//import Adapter from 'enzyme-adapter-react-16';
//Redux test store
import configureStore from 'redux-mock-store'
import Footer from './Footer.js';

const initialState = { VisibilityFilters: VisibilityFilters.SHOW_ALL };
const mockStore = configureStore()

let store, container

store = mockStore(initialState)
//Enzyme.configure({ adapter: new Adapter() })

test('renders correctly', () => {

    //const footer = shallow(<Provider store={store}><Footer /></Provider>)
    //let footerView = footer.toJSON();
    //expect(footer).toMatchSnapshot();

    //footer.first('a').simulate('click');
    //expect(footer).toMatchSnapshot();

    const footer = renderer
        .create(<Provider store={store}><Footer /></Provider>);

    let footerView = footer.toJSON();
    expect(footerView).toMatchSnapshot();

});

test('filter link', () => {

    const filterLink = renderer.create(<Provider store={store}><FilterLink filter={VisibilityFilters.SHOW_ALL} onClick={Function}>Active</FilterLink></Provider>);
    let filterLinkView = filterLink.toJSON();
    filterLinkView.props.onClick({
        preventDefault: () => {
        }
    });
    filterLinkView = filterLink.toJSON();
    expect(filterLinkView).toMatchSnapshot();
});





