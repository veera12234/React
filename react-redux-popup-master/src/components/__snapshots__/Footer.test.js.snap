// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter link 1`] = `
<a
  href="#"
  onClick={[Function]}
>
  Active
</a>
`;

exports[`renders correctly 1`] = `
<p>
  Show:
   
  <a
    href="#"
    onClick={[Function]}
  >
    All
  </a>
  , 
  <a
    href="#"
    onClick={[Function]}
  >
    Active
  </a>
  , 
  <a
    href="#"
    onClick={[Function]}
  >
    Completed
  </a>
</p>
`;
